src/app/(default)/page.tsx

```diff
@@ -12,37 +11,30 @@ export default async function Index({
 }) {
   const currentPageNumber = searchParams.page ? Number(searchParams.page) : 1
 
-  const { posts, postCount } = await api.post.feed.query({
-    take: POSTS_PER_PAGE,
-    skip:
-      currentPageNumber === 1
-        ? undefined
-        : POSTS_PER_PAGE * (currentPageNumber - 1),
+  const cachedData = cache(async () => {
+    return await api.post.feed.query({
+      take: POSTS_PER_PAGE,
+      skip:
+        currentPageNumber === 1
+          ? undefined
+          : POSTS_PER_PAGE * (currentPageNumber - 1),
+    })
```

Comment:
How long does this query take? If it's slow, we should fix the query before caching it.

Reply:
It's coming in about 490ms, so pretty long...

Reply:
Great info. I checked insights and this query looks like it's max taking 5-10ms. Which is slow for that query & data size, so can likely be optimized. But that's not the source of the 490ms.

Next places to look:
- Where is the server vs db? Is it network latency?
- How is the connection established? Is there anything in prisma in between?

Reply:
Great, I'll keep digging in, get it fixed up.

Reply:
@ayrton @mscoutermarsh I've done a fair amount of work here to try and get this faster, and have also improved some of the UI using the initial data.

On the query latency, I'm not seeing anything happening on the Prisma end that would lead to higher latency like this; in some testing, caching the query calls seemed to have done quite a bit to improve load times across the board. Overall, the Prisma and data implementations are exactly the same as the old one, but in the 2.0 we're getting the data on the server rather than the client, and we're removing loading skeletons, which we had before. Since that would block the TTFB, I think it could be leading to us noticing a slower initial load. Any suggestions for getting this stronger are welcome, I'll try and get them implemented over the weekend.